apps/web/src/data-access/workspaceOnboarding.ts

```diff
@@ -59,3 +60,17 @@ export async function getOnboardingResources() {
 
   return { project, commit }
 }
+
+export async function getNecessaryOnboardingSteps() {
+  const { workspace } = await getCurrentUserOrRedirect()
+  if (!workspace?.id) {
+    throw new NotFoundError('Workspace ID is required')
```

Comment:
Use next.js `notFound`
https://nextjs.org/docs/app/api-reference/functions/not-found